03_letters/luliangLi/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+type Pair struct {
+	key   rune
+	value int
+}
+
+func (p Pair) String() string {
+	return fmt.Sprintf("%s:%d", string(p.key), p.value)
+}
+
+func letters(s string) map[rune]int {
+	rlt := make(map[rune]int)
+
+	for _, char := range s {
+		val := rlt[char]
+		rlt[char] = val + 1
+	}
+
+	return rlt
+}
+
+func sortLetters(m map[rune]int) []string {
+	pairs := make([]Pair, len(m))
```

Comment:
What is the main purpose of using the `Pair` struct that you have created? 
And could you just use the `map[rune]int` for the same purpose?

Reply:
Agree with @CameronSchafer here, using the `Pair` struct means you duplicate the looping logic. You can achieve the same outcome by directly creating a `[]string` from the `map[rune]int`.
```go
for k, v := range m {
  output = append(output, fmt.Sprintf("%c:%d", k, v))
}
```